docs/time_crystals/time_crystal_circuit_generation.ipynb

```diff
@@ -0,0 +1,160 @@
+    "\n",
+    "The quantum computer supplies a system of many qubits, locally connected to each other in a chain. A many-body local system like this is critical for the existence of a time crystal. Without an MBL system, it is expected that the system's state would decay into a maximum entropy state that is incompatible with the goal of stable and consistent time structure. \n",
+    "\n",
+    "The time-crystalline behavior that the DTC circuits demonstrate is perhaps the simplest kind of time-structured behavior, oscillation. Each circuit is built with some number of identical $U$-cycles. Time is represented by a circuit list where each circuit is ordered with increasingly many $U$-cycles; each cycle is considered a discrete time step. The eventual effect of these $U$-cycles is consistent oscillations of each qubits' polarizations. The experiments performed demonstrate that this time-crystalline oscillation behavior is stable in spite of different initial states and introduced random potentials. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "f53722cb0850"
+   },
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "0e4595827ec0"
+   },
+   "outputs": [],
+   "source": [
+    "!pip install cirq --quiet\n",
+    "try:\n",
+    "    import recirq\n",
+    "except ImportError:\n",
+    "    !pip install --quiet git+https://github.com/quantumlib/ReCirq"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "e5e4f66e8e67"
+   },
+   "outputs": [],
+   "source": [
+    "import cirq\n",
+    "import recirq.time_crystals as time_crystals"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "a12b700f009e"
+   },
+   "source": [
+    "## Circuit Construction\n",
+    "Each DTC circuit is created with symbolic parameters. Parameter values are supplied near run/simulation time with a `cirq.ParamResolver`, which means the circuit list needs to be generated only once for potentially many different experimental parameter configurations. \n",
+    "\n",
+    "The code below uses an IPython-specific utility to inspect the code of the key function that creates the symbolic circuit list, `recirq.time_crystals.symbolic_dtc_circuit_list()`. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "c0f1c1475908"
+   },
+   "outputs": [],
+   "source": [
+    "import inspect\n",
+    "from IPython.display import Code\n",
+    "\n",
+    "Code(inspect.getsource(time_crystals.symbolic_dtc_circuit_list), language=\"python\")"
```

Comment:
This seems weird to me that we are printing out the code as an output of the colab.  I'll leave the decision up to you if you stil think this way is clearer, but I would consider linking to the source code with text such as "The code, which can be viewed here (link) is surprisingly succinct."

Reply:
My intent with printing the code was to provide a detailed example of how the circuits are constructed. Perhaps a couple step-by-step examples of each moment for a circuit of a few qubits would be a better example. I'll try this and see how it looks.

Reply:
I think it's kinda neat 🤷